app/src/lib/editors/win32.ts

```diff
@@ -159,11 +157,26 @@ function extractApplicationInformation(
     return { displayName, publisher, installLocation }
   }
 
-  if (
-    editor === ExternalEditor.VisualStudioCode ||
-    editor === ExternalEditor.SublimeText
-  ) {
+  if (editor === ExternalEditor.VisualStudioCode) {
+    for (const item of keys) {
+      if (item.name === 'DisplayName') {
+        displayName = item.value
+      } else if (item.name === 'Publisher') {
+        publisher = item.value
+      } else if (item.name === 'Inno Setup: App Path') {
+        installLocation = item.value
+      }
+    }
+
+    return { displayName, publisher, installLocation }
+  }
+
+  if (editor === ExternalEditor.SublimeText) {
     for (const item of keys) {
+      // NOTE:
+      // Sublime Text bakes the build number into the DisplayName value, so for
+      // forward-compatibility whenever they decide to drop a new build let's
+      // use this entry which doesn't contain the version number
       if (item.name === 'Inno Setup: Icon Group') {
```

Comment:
I have absolutely no clue about how Inno Setup but https://github.com/desktop/desktop/issues/3304#issuecomment-344340564 mentions that Icon Group could be affected by user input which seems bad. That comment was made about VS Code but if that applies to Sublime Text as well could we use `displayName` but check for a prefix match instead? I.e.

```ts
else if (item.name === 'DisplayName' && /^Sublime Text/.test(item.value)) {
    displayName = 'Sublime Text'
}

Reply:
Or `String.prototype.startsWith()`?

Reply:
> Or `String.prototype.startsWith()`?

Yeah, I just didn't know if `item.value` could be null/undefined. The RegEx route would work regardless of what we pass it whereas `startsWith` _could_ need an additional check. I'm down for either or.

```ts
else if (item.name === 'DisplayName' && item.value && item.value.startsWith('Sublime Text')) {
    displayName = 'Sublime Text'
}
```

Reply:
Yeah, I'll suck it up and be more tolerant of the Sublime Text values 👍